pkg/cloud/services/network/network.go

```diff
@@ -99,8 +105,20 @@ func (s *Service) DeleteNetwork() (err error) {
 
 	vpc.DeepCopyInto(s.scope.VPC())
 
+	// VPC Endpoints.
+	conditions.MarkFalse(s.scope.InfraCluster(), infrav1.VpcEndpointsReadyCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
+	if err := s.scope.PatchObject(); err != nil {
+		return err
+	}
+
+	if err := s.deleteVPCEndpoints(); err != nil {
+		conditions.MarkFalse(s.scope.InfraCluster(), infrav1.VpcEndpointsReadyCondition, "DeletingFailed", clusterv1.ConditionSeverityWarning, err.Error())
+		return err
+	}
+	conditions.MarkFalse(s.scope.InfraCluster(), infrav1.VpcEndpointsReadyCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
+
 	// Routing tables.
-	conditions.MarkFalse(s.scope.InfraCluster(), infrav1.RouteTablesReadyCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
+	conditions.MarkFalse(s.scope.InfraCluster(), infrav1.VpcEndpointsReadyCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
This looks like a mistake

Reply:
Fixed.

To your comment above, the correct link to understand why this is helpful is https://docs.aws.amazon.com/vpc/latest/privatelink/gateway-endpoints.html — I added this to the comment. While traffic doesn't require NAT gateways, the overall benefit is saving egress/ingress because traffic stays within AWS. 

In the future, I'd expect we could setup more VPC endpoints, or give the user to set them up with well known endpoints.

> The benefits should be substantial since we need quite a few new permissions.
I generally agree with the sentiment that permissions should not be taken lightly, that said, new features, improvements, or general enhancements shouldn't shy away to be implemented even if they need new permissions. CAPA is more useful if it offers a wide range of features, and has sane defaults.